src/clients/inst_add-on.rb

```diff
@@ -68,18 +69,16 @@ def main
           ret = NetworkSetupForAddons(missing_addons)
           return ret if Builtins.contains([:back, :abort], ret)
 
-          begin
-            tmp = Tempfile.new("cmdline-addons-")
-            # each add-on on a separate line
-            File.write(tmp.path, missing_addons.join("\n"))
-            # import the add-ons from the temporary file
-            AddOnProduct.AddPreselectedAddOnProducts(
-              [{ "file" => tmp.path, "type" => "plain" }]
-            )
-          ensure
-            tmp.close
-            tmp.unlink
+          plaindir, download, name, alias_ = false, true, "", ""
```

Comment:
I expect you want to have it only for marking so I prefer in such case having it directly in call like

```
createSourceImpl(url, plaindir = false, download = true, name = "", alias_ = "")
```

Reply:
Yes, but guess what: "warning: assigned but unused variable - plaindir ..."